h2o-core/src/main/java/hex/ParallelModelBuilder.java

```diff
@@ -0,0 +1,56 @@
+package hex;
+
+import jsr166y.ForkJoinTask;
+import java.util.*;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+public class ParallelModelBuilder extends ForkJoinTask<ParallelModelBuilder> {
```

Comment:
This should be moved to autoML module probably. Only a showcase, but fully functional. Will be instantiated inside a method and won't be serialized anywhere.